File: typescript/src/data_models/event/event.ts
import {EventType} from './event_type';

/**
 * Describes a triggerable event.
 */

export class Event {
  /**
   * The Model ID of the Event as it is declared in the ProcessModel.
   */
  public id: string;
  /**
   * The name of the Event as it is declared in the ProcessModel.
   */
  public eventName: string;
  /**
   * The BPMN type of the event (StartEvent, EndEvent, etc).
   */
  public bpmnType: string;
  /**
   * The type of the event (Message, Signal, etc).
   */
  public eventType: EventType;
  /**
   * The Instance ID of the Event.
   */
  public flowNodeInstanceId?: string;
  /**
   * The ID of the correlation that the Event belongs to.
   */
  public correlationId: string;
  /**
   * The ID of the ProcessModel that the Event belongs to.
   */
  public processModelId: string;
  /**
   * The ID of the ProcessInstance that the Event belongs to.
   */
  public processInstanceId?: string;
}

File: dotnet/src/data_models/event/Event.cs
namespace ProcessEngine.ConsumerAPI.Contracts.DataModel
{
     /// <summary>
    /// Describes an event that can be triggered through the ConsumerAPI.
    /// </summary>
    public class Event
    {
        /// <summary>
        /// The model ID of the event, as it is declared in the ProcessModel.
        /// </summary>
        public string Id { get; set; }

        /// <summary>
        /// The name of the event, as it is declared in the ProcessModel.
        /// </summary>
        public string EventName { get; set; }

        /// <summary>
        /// The BPMN type of the event (StartEvent, EndEvent, etc).
        /// </summary>
        public string BpmnType { get; set; }

        /// <summary>
        /// The type of the event (Message, Signal, etc).
        /// </summary>
        public string EventType { get; set; }

        /// <summary>
        /// The instance ID of the event.
        /// </summary>
        public string FlowNodeInstanceId { get; set; }

        /// <summary>
        /// The corresponding id of the Correlation holding the event.
        /// </summary>
        public string CorrelationId { get; set; }

        /// <summary>
        /// The corresponding id of the ProcessModel.
        /// </summary>
        public string ProcessModelId { get; set; }

        /// <summary>
        /// The id of the ProcessInstance.
        /// </summary>
        public string ProcessInstanceId { get; set; }
    }
}
